### DISCUSSION TITLE
First flet app - please test and review

### DISCUSSION THREAD

**Discussion by clstaudt** [flet-dev/flet#363] [Show and tell]
Hello flet.dev community, I have just released my first demo app built with flet. I'd be grateful for some testing feedback in order to decide whether flet is the way to go.

https://github.com/tuttle-dev/tuttle/releases/tag/v0.1.0

I am still having difficulties getting the packaged app to run on Windows, therefore this first release is for macOS and Linux.
![Screenshot 2022-09-20 at 22 27 30](https://user-images.githubusercontent.com/875194/191358088-825d59d7-55b7-4e8b-b22c-cc99c561f819.png)
![Screenshot 2022-09-20 at 22 29 38](https://user-images.githubusercontent.com/875194/191358334-c684b154-3d18-46db-8268-fb8216c67d20.png)
![Screenshot 2022-09-20 at 22 27 53](https://user-images.githubusercontent.com/875194/191358149-54860e2f-3b5f-4e65-9dd1-c08fb82db5a9.png)

**Comment by FeodorFitsner** [comment]
Cool! Could you share a live demo/video/screenshots please?

**Reply by modaye** [reply]
Is there a way to display icons on the right

**Comment by jet10000** [comment]
<img width="412" alt="image" src="https://user-images.githubusercontent.com/2258120/191426513-5ad66ff6-664e-4555-bde1-f3db257270fb.png">

**Reply by clstaudt** [reply]
@jet10000  Are you running on an Intel Mac?

**Reply by jet10000** [reply]
yes

**Reply by clstaudt** [reply]
@jet10000  Since I don't own one I'm still looking for a way to build for x86 Mac. Perhaps https://www.appveyor.com?

**Reply by modaye** [reply]
yes, you need ci

**Reply by clstaudt** [reply]
I am aware of that, looking for experience-based recommendations.

**Reply by modaye** [reply]
Ha ha ha, here's the answer https://github.com/pyinstaller/pyinstaller/issues/6296

**Reply by clstaudt** [reply]
That looks promising. I don't see yet how it handles x86 vs arm, do you @modaye? The last comment gives some remarks on why it might not be so simple.

**Comment by modaye** [comment]
Looking forward to an online example

**Reply by clstaudt** [reply]
@modaye It's a desktop app, I am mainly interested to hear about the experience of running it locally. What would be the benefit of an online example?

**Reply by modaye** [reply]
It's great to be able to experience the project through the browser

**Reply by clstaudt** [reply]
It's somewhat contrary to the idea of the app (no intention to keep user data on a server), but I'll think about it.

**Comment by ghost** [comment]
How did you manage to navigate through the pages using navigationRail

**Reply by clstaudt** [reply]
@martin265  I adapted an example from the flet-examples repo: https://github.com/flet-dev/examples/tree/main/python/community/responsive_menu_layout

... into a simpler version (aimed at the desktop, without so much code for responsiveness): https://github.com/flet-dev/examples/tree/main/python/community/simple_desktop_layout

That said, I am a data scientist, not a UI/UX developer. I wish the flet project would provide more of these best practices examples of how to build non-trivial UIs and layouts (FYI @FeodorFitsner).

**Comment by clstaudt** [comment]
Mac users (ARM or Intel) - can I ask you to try and run these app binaries again?
https://github.com/tuttle-dev/tuttle/releases/tag/v0.2.1

@jet10000

**Reply by ndonkoHenri** [reply]
Not available for Windows?

Or you deliberately don't plan to support that..¿

(I wanted to test🙂)

**Reply by clstaudt** [reply]
@ndonkoHenri   Here you go, I managed to build it for Windows and updated the release.

Looks like there are several bugs in the Windows version that don't appear in the macOS version. For example, the NavigationRail is not always updating the main window area as intended. That's confusing.

Feel free to test.

**Reply by ndonkoHenri** [reply]
Ok. I will give it a try when possible.

Did you build it using Appveyor? Or yourself using pyinstaller..?

**Reply by clstaudt** [reply]
@ndonkoHenri  I built it with GitHub actions: https://github.com/tuttle-dev/tuttle/blob/main/.github/workflows/pyinstaller-build.yml

I haven't figured out Appveyor yet, and I am not sure which advantages it will bring over GitHub actions.

**Comment by jet10000** [comment]
<img width="698" alt="image" src="https://user-images.githubusercontent.com/2258120/196580554-24b35918-65fe-4d1a-9d3e-8f3aae3f6d44.png">

**Reply by clstaudt** [reply]
@jet10000  fixed
